### PR TITLE
Enable Sauce runs on IE on an opt-in basis

### LINKED ISSUE #10731: Support running _some_ integration tests on IE11
We've had a few IE11 bugs, e.g. amp-bind, amp-ad, amp-carousel: https://github.com/ampproject/amphtml/issues?utf8=%E2%9C%93&q=is%3Aissue%20IE%2011

Those bug fixes don't have any regression protection since [we don't run Saucelabs tests on IE11](https://github.com/ampproject/amphtml/blob/4a9419322d1fa42e6a65c77dd2c35d32bc5a659b/build-system/tasks/runtime-test.js#L69). 

Let's fix that by supporting IE11 on Saucelabs and allow certain tests to opt-in (**disabled by default**). 

/cc @erwinmombay @rsimha-amp

### PR DESCRIPTION
**Pull request by rsimha-amp**
This PR does the following:
1. Adds IE 11 to the mix of browsers used on sauce labs for testing
2. Disables testing on IE 11 by default
3. Adds functions `enableIe` and `ifIe` to allow test writers to opt in to testing on IE
4. Updates all integration tests that use just `describe(` with `describe().configure().run(`, so that the new default `TestConfig` is picked up

In the new world:
- `describe('foo')` is no longer meant to be used for integration tests (if used, it will include IE)
- `describe().configure().run()` will run tests on all browsers except IE
- `describe().configure().enableIe().run()` will run tests on all browsers including IE
- `describe().configure().ifIe().run()` will run tests only on IE

Fixes #10731

### CODE REVIEW

File: test/_init_tests.js [hunk 1]
```diff
@@ -160,6 +166,10 @@ class TestConfig {
     return this.if(this.platform.isIos.bind(this.platform));
   }
 
+  ifIe() {
+    return this.if(this.platform.isIe.bind(this.platform));
```
Comment (choumx) [inline test/_init_tests.js]: Not sure this will work. Probably need to remove the `skipIe()` function from `skipMatchers` array.

Comment (jridgewell) [inline test/_init_tests.js]: Yup.

Comment (rsimha-amp) [inline test/_init_tests.js]: Isn't this just a function that can be used by a test writer to run a test exclusively on IE, should they choose to? The way I understood this class, the default for `describe()` is to run tests on all platforms except those on which we call `skipFoo`, and the behavior for `ifFoo` is to run that test only on `Foo` browser. 

I'm still testing this out, so I could be wrong.

Comment (choumx) [inline test/_init_tests.js]: That's correct, but we haven't had any platforms skipped by default yet. Check out the implementation of `this.if()`.

Comment (rsimha-amp) [inline test/_init_tests.js]: I thought the line I added to the constructor to skip IE by default (line 104) would do what we wanted. I'll re-read the code to see why that isn't the case.

**Comment by rsimha-amp** [comment]
Opted in amp-bind tests.

File: test/_init_tests.js (hunk 1 above)
Comment (rsimha-amp) [inline test/_init_tests.js]: Aha, the lists are independent, so we need to remove it from `skipMatchers` before `ifIe` kicks in. Done.

File: test/_init_tests.js [hunk 2]
```diff
@@ -160,6 +166,15 @@ class TestConfig {
     return this.if(this.platform.isIos.bind(this.platform));
   }
 
+  ifIe() {
+    // Necessary because we skip IE by default.
+    const ieSkipped = this.skipMatchers.indexOf(this.skipIe);
```
Comment (choumx) [inline test/_init_tests.js]: Does this work? I think `skipMatchers` contains `this.platform.isIe.bind(this.platform)`, not `this.skipIe`. Probably want to save the former as an ivar and use it in both `skipIe()` and here.

**Comment by rsimha-amp** [comment]
@choumx, we now have a working solution for sauce labs testing on IE. See PR description for details.

I've gone ahead and enabled a few integration tests like `amp-bind` on IE to see if they pass or not.

**Comment by rsimha-amp** [comment]
@choumx, @cramforce, we now have IE 11 in our browser matrix. I've enabled a bunch of tests on IE 11: see https://travis-ci.org/ampproject/amphtml/jobs/267750202

This PR is ready for review.

File: extensions/amp-bind/0.1/test/integration/test-bind-impl.js [hunk 3]
```diff
@@ -99,7 +99,7 @@ function waitForEvent(env, name) {
   });
 }
 
-describe.configure().ifNewChrome().run('Bind', function() {
+describe.configure().ifIe().ifNewChrome().run('Bind', function() {
```
Comment (choumx) [inline extensions/amp-bind/0.1/test/integration/test-bind-impl.js]: Should this be `enableIe()`?

File: test/integration/test-amp-bind.js [hunk 4]
```diff
@@ -71,7 +71,7 @@ describe.configure().ifNewChrome().run('amp-bind', function() {
     return fixture.awaitEvent(BindEvents.RESCAN_TEMPLATE, ++numTemplated);
   }
 
-  describe('with [text] and [class]', () => {
+  describe.configure().run('with [text] and [class]', () => {
```
Comment (choumx) [inline test/integration/test-amp-bind.js]: Are the changes in this file necessary? The top-level describe in this file already has `configure()`.

File: test/_init_tests.js [hunk 5]
```diff
@@ -128,6 +133,12 @@ class TestConfig {
     return this.skip(this.platform.isIos.bind(this.platform));
   }
 
+  enableIe() {
+    this.skipMatchers.splice(
+        this.skipMatchers.indexOf(this.platform.isIe.bind(this.platform)), 1);
```
Comment (jridgewell) [inline test/_init_tests.js]: This will not work. `Function#bind` returns a new function instance every time.

File: test/integration/test-visibility-states.js [hunk 6]
```diff
@@ -123,8 +123,8 @@ describe.configure().ifNewChrome().run('Viewer Visibility State', () => {
       });
     });
 
-    describe('from in the PRERENDER state', () => {
-      describe('for prerenderable element', () => {
+    describe.configure().run('from in the PRERENDER state', () => {
+      describe.configure().run('for prerenderable element', () => {
```
Comment (choumx) [inline test/integration/test-visibility-states.js]: Ditto. Don't we only need a `configure()` on the top-level describe?

File: extensions/amp-bind/0.1/test/integration/test-bind-impl.js (hunk 3 above)
Comment (rsimha-amp) [inline extensions/amp-bind/0.1/test/integration/test-bind-impl.js]: Hmm. I did some more testing, and it turns out that due to how `ifFoo` fully replaces the platform on which tests are run, chaining together `ifIe` and `ifChrome`, for example, does not have the effect we desire. Chaining together `enableIE` and `ifChrome` doesn't work either. The only way to run a test on more than one platform is to skip all the others, or to create a whole set of `enableFoo` methods that can then be chained together. Taking another look to see if there's a less invasive way of making this work.

File: test/integration/test-amp-bind.js (hunk 4 above)
Comment (rsimha-amp) [inline test/integration/test-amp-bind.js]: Good point. Reverted.

File: test/integration/test-visibility-states.js (hunk 6 above)
Comment (rsimha-amp) [inline test/integration/test-visibility-states.js]: Reverted.

File: extensions/amp-bind/0.1/test/integration/test-bind-impl.js (hunk 3 above)
Comment (rsimha-amp) [inline extensions/amp-bind/0.1/test/integration/test-bind-impl.js]: Reverted this file, as the change was unnecessary. Enabling this test on IE and on Chrome will require us to skip all other platforms. Happy to add that code here, or do it in another PR.

Comment (choumx) [inline extensions/amp-bind/0.1/test/integration/test-bind-impl.js]: Yea, we could prob do this with something like `describe.configure().on(['ie', 'new-chrome'].run(...)`.

**Comment by rsimha-amp** [comment]
All comments are addressed, and we now have more than a dozen tests running and passing on IE 11. Merging this PR.
